protocols/v2/binary-sv2/no-serde-sv2/codec/src/codec/mod.rs

```diff
@@ -13,52 +63,107 @@ use buffer_sv2::Slice;
+///     be determined with respect to the current instance of the type.
+///
+/// These methods are crucial in decoding scenarios where the full size of the message
+/// is not immediately known, helping to determine how many bytes need to be read.
 pub trait SizeHint {
+    /// `size_hint` is a static method that takes the raw data and an offset and returns the total
+    /// size of the encoded message. This is particularly useful for types where the encoded size
+    /// may vary based on the contents of the data, and we need to calculate how much space is
+    /// required for decoding.
     fn size_hint(data: &[u8], offset: usize) -> Result<usize, Error>;
+    /// `size_hint_` is an instance method that performs the same function but allows the size to be
+    /// be determined with respect to the current instance of the type. 
     fn size_hint_(&self, data: &[u8], offset: usize) -> Result<usize, Error>;
 }
 
-/// Return the encoded byte size of an `Encodable` comprehensive of the header, if any
+/// The `GetSize` trait returns the total size in bytes of an encodable type.
+///
+/// It provides a single method, `get_size`, which returns the total size of the type
+/// in bytes.
 pub trait GetSize {
+    /// `get_size` returns total size of the type in bytes.
     fn get_size(&self) -> usize;
 }
 
 #[cfg(feature = "with_buffer_pool")]
 impl GetSize for Slice {
+    // Provides the total size of a `Slice` by returning its length.
+    //
+    // This implementation for the `Slice` type returns the number of bytes in the
+    // slice, which represents its total size when encoded.
     fn get_size(&self) -> usize {
         self.len()
     }
 }
-
-/// Implemented by all the primitives with a fixed size
+/// The `Fixed` trait is implemented by all primitives with a constant, fixed size.
+///
+/// Types implementing this trait must define the constant `SIZE`, representing the
+/// fixed number of bytes needed to encode or decode the type. This trait is used for
+/// types that have a know size at compile time , such as integers, fixed-size arrays, etc.
 pub trait Fixed {
+    ///the constant `SIZE`, represents the fixed number of bytes needed to encode or decode the type.
     const SIZE: usize;
 }
 
+/// The `Variable` trait is designed for types that have variable size when encoded.
+///
+/// Types implementing this trait provide the following:
+///
+/// - `HEADER_SIZE`: The size of the header in bytes. This header often contains metadata like
+///     the length of the variables-sized data.
+/// - `MAX_SIZE`: The maximum allowed size for this type. This value is essential in
+///     ensuring that dynamically sized data does not exceed defined limits.
+///
+/// The trait also includes methods to calculate the inner size of the data (`inner_size`) and
+/// to return the header as a byte vector (`get_header`). These methods are essential for
+/// managing dynamically sized types in scenarios like serialization and encoding.
```

Comment:
I cannot find `Variable` anywhere in the Rust Doc rendering

it seems this is not really public

Reply:
No its not, we are not even using it. I added removal  of it in refactor